common/src/main/java/com/revenuecat/purchases/common/OfferingFactories.kt

```diff
@@ -49,15 +49,25 @@ fun JSONObject.createOffering(productsById: Map<String, StoreProduct>): Offering
     }
 }
 
+fun chooseBestOffer(products: List<StoreProduct>?): StoreProduct? {
+    //TODO this needs to be improved
+    if (products == null)
+        return null
+
+    return products.maxWithOrNull(compareBy { it.pricingPhases?.size })
+}
+
 fun JSONObject.createPackage(
-    productsById: Map<String, StoreProduct>,
+    productsById: Map<String, List<StoreProduct>>,
     offeringIdentifier: String
 ): Package? {
-    val sku = getString("platform_product_identifier")
-    return productsById[sku]?.let { product ->
+    val sku = optString("platform_product_group_identifier")
+    val product_identifier = getString("platform_product_identifier")
```

Comment:
Is `platform_product_group_identifier` part of the new backend response? I'm wondering, what's the difference between `platform_product_group_identifier` and `platform_product_identifier`. I'm guessing `platform_product_group_identifier` is the subscription id and the `platform_product_identifier` is the base plan id but not completely sure.

Reply:
More or less - here's some [more info](https://docs.google.com/document/d/1l1gkp0IAR2tOqM6msTytucOxBy9sNGt31bpL_W-mNkQ/edit#heading=h.21rdbt1jkmik). Unfortunately we can't match the base_plan_id, because they're not exposed by BC. We opted for matching products to baseplans based on the duration (another new field we added).